library/lua/gui.lua

```diff
@@ -706,27 +706,41 @@ ZScreen.ATTRS{
 
 function ZScreen:init()
     self.saved_pause_state = df.global.pause_state
-    if self.initial_pause then
+    if self.initial_pause and dfhack.isMapLoaded() then
         df.global.pause_state = true
     end
     self.defocused = false
 end
 
 function ZScreen:dismiss()
     ZScreen.super.dismiss(self)
-    if self.force_pause or self.initial_pause then
+    if (self.force_pause or self.initial_pause) and dfhack.isMapLoaded() then
         -- never go from unpaused to paused, just from paused to unpaused
         df.global.pause_state = df.global.pause_state and self.saved_pause_state
     end
 end
 
+local NO_LOGIC_SCREENS = utils.invert{
+    'viewscreen_loadgamest',
+    'viewscreen_export_regionst',
+    'viewscreen_choose_game_typest',
+    'viewscreen_worldst',
+}
```

Comment:
I would tend to prefer to verify that these names are correct by looking up the classes, e.g. with
```lua
for k, v in pairs(NO_LOGIC_SCREENS) do
    NO_LOGIC_SCREENS[k] = df[v] or error('invalid class name: ' .. v)
end
```

and then replacing `vs_name = getmetatable(...)` later with `vs_type = getDFViewscreen()._type`